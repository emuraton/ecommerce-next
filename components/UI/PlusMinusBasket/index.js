// @flow
import * as React from 'react';
import styled from 'styled-components';
import { ifProp } from 'styled-tools';

import PlusIcon from '../SVG/PlusIcon';
import MinusIcon from '../SVG/MinusIcon';

const TicketAction = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  padding-left: 20px;
`;

const Button = styled.button`
  padding: 0;
  border: 0;
  border-radius: 0;
  cursor: pointer;
  font-size: 0;
  background-color: white;
  transition: transform 0.2s;
  outline: 0;
  opacity: ${ifProp('isDisabled', 0.2, 1)};

  &:active {
    transform: scale(0.95);
  }
`;
Button.displayName = 'Button';

const Quantity = styled.span`
  margin: 0 25px;
  margin-top: 5px;
  color: rgb(0, 132, 137);
  opacity: ${ifProp('isDisabled', 0.2, 1)};
  font-size: 26px;
`;
Quantity.displayName = 'Quantity';

const StyledPlusIcon = styled(PlusIcon)`
  height: 30px;
  width: 30px;
  fill: rgb(0, 132, 137);
`;

const StyledMinusIcon = styled(MinusIcon)`
  height: 30px;
  width: 30px;
  fill: rgb(0, 132, 137);
`;

type Props = {
  onClick: (event: SyntheticMouseEvent<any>) => {},
  quantity: number,
};

const PlusMinusBasket = ({ onClick, quantity }: Props) => (
  <TicketAction>
    <Button element="button" onClick={onClick} isDisabled={quantity === 6}>
      <StyledPlusIcon />
    </Button>
    <Quantity isDisabled={quantity === 0}>{quantity}</Quantity>
    <Button element="button" onClick={onClick} isDisabled={quantity === 0}>
      <StyledMinusIcon />
    </Button>
  </TicketAction>
);

export default PlusMinusBasket;
