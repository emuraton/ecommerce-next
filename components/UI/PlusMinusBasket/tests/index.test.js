import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';

import PlusMinusBasket from '..';

describe('<PlusMinusBasket />', () => {
  describe('@renders', () => {
    const initialProps = {
      onClick: Function.prototype,
      quantity: 0,
    };

    it('default render', () => {
      const wrapper = shallow(<PlusMinusBasket {...initialProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();

      const buttons = wrapper.find('Button');
      expect(buttons.at(0)).toHaveStyleRule('opacity', '1');
      expect(buttons.at(1)).toHaveStyleRule('opacity', '0.2');

      const quantity = wrapper.find('Quantity');
      expect(quantity).toHaveStyleRule('opacity', '0.2');
    });

    it('with quantity > 0', () => {
      const props = {
        ...initialProps,
        quantity: 1,
      };
      const wrapper = shallow(<PlusMinusBasket {...props} />);
      expect(toJson(wrapper)).toMatchSnapshot();

      const buttons = wrapper.find('Button');
      expect(buttons.at(0)).toHaveStyleRule('opacity', '1');
      expect(buttons.at(1)).toHaveStyleRule('opacity', '1');

      const quantity = wrapper.find('Quantity');
      expect(quantity).toHaveStyleRule('opacity', '1');
    });
  });
});
