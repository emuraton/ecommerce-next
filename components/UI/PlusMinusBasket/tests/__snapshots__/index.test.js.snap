// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PlusMinusBasket /> @renders default render 1`] = `
<PlusMinusBasket__TicketAction>
  <Button
    element="button"
    isDisabled={false}
  >
    <PlusMinusBasket__StyledPlusIcon />
  </Button>
  <Quantity
    isDisabled={true}
  >
    0
  </Quantity>
  <Button
    element="button"
    isDisabled={true}
  >
    <PlusMinusBasket__StyledMinusIcon />
  </Button>
</PlusMinusBasket__TicketAction>
`;

exports[`<PlusMinusBasket /> @renders with quantity > 0 1`] = `
<PlusMinusBasket__TicketAction>
  <Button
    element="button"
    isDisabled={false}
  >
    <PlusMinusBasket__StyledPlusIcon />
  </Button>
  <Quantity
    isDisabled={false}
  >
    1
  </Quantity>
  <Button
    element="button"
    isDisabled={false}
  >
    <PlusMinusBasket__StyledMinusIcon />
  </Button>
</PlusMinusBasket__TicketAction>
`;
